### PR TITLE
Yet another PR for Julia v1

### PR DESCRIPTION
**Pull request by jaakkor2**
Take master, merge #86, resolve conflicts and remove Compat.

Status 2019-2-18:
- Travis passes [build job 204](https://travis-ci.org/JuliaIO/MAT.jl/builds/494832827)
- local testing `]test MAT`
    - windows 7 x64 Julia v1.1 passes

### CODE REVIEW

File: src/MAT_HDF5.jl [hunk 1]
```diff
@@ -296,12 +292,12 @@ elseif length(s) > 63
 end
 
 toarray(x::Array) = x
-toarray(x::Array{Bool}) = reinterpret(UInt8, x)
+toarray(x::Array{Bool}) = reinterpret.(UInt8, x)
```
[the remark this answers is not in the document]
Comment (jaakkor2) [reply src/MAT_HDF5.jl]: True. Reverted back.

File: REQUIRE [hunk 2]
```diff
@@ -1,5 +1,4 @@
-julia 0.6
-HDF5
+julia 0.7
```
[the remark this answers is not in the document]
Comment (jaakkor2) [reply REQUIRE]: I was following HDF5,  but now added 1.1.  Seems Travis results are not consistent from run to run. Would be good to allow installation on Julia v0.7 if it helps in transition from v0.6 to v1.x

File: src/MAT_v5.jl [hunk 3]
```diff
@@ -154,6 +157,24 @@ function read_data(f::IO, swap_bytes::Bool, ::Type{T}, dimensions::Vector{Int32}
...
+    end
+    if read_array
+        data = read_bswap(f, swap_bytes, reinterpret(read_type, d))
+    else
+        data = read_bswap(f, swap_bytes, read_type)
+    end
+    skip_padding(f, nbytes, hbytes)
+
+    read_array ? d : convert(T, data)
+end
```
[the remark this answers is not in the document]
Comment (jaakkor2) [reply src/MAT_v5.jl]: This came from PR #86, commit https://github.com/halleysfifthinc/MAT.jl/commit/d922093d93ac956adc6de3d13d36bf35e175aa74 . I would recommend filing a separate issue for it.

File: src/MAT_HDF5.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaakkor2) [reply src/MAT_HDF5.jl]: The concern is addressed as far as I can see.
```
using BenchmarkTools; a = rand(Bool, 2^20);
@btime reinterpret.(UInt8, $a);
 346.727 μs (2 allocations: 1.00 MiB)
 @btime reinterpret(UInt8, $a);
  8.032 ns (1 allocation: 32 bytes)
```

File: src/MAT_v5.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaakkor2) [reply src/MAT_v5.jl]: Since this PR is already too massive, I would prefer solving this concern in another PR.

**Comment by alexcanepa** [comment]
@yuyichao bump on this PR, it's pretty frustrating to depend on forks of this package for usability

File: src/MAT_v5.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuyichao) [reply src/MAT_v5.jl]: I mean, replacing old method is fine, but I really can't find where is this method even used. There's no caller of read_data with three argument in the current repo and this PR doesn't seem to add any functions. In another word, this is a request to delete this method if it doesn't affect test result.

File: src/MAT_HDF5.jl [hunk 4]
```diff
@@ -607,14 +603,14 @@ function read(obj::HDF5Object, ::Type{Bool})
     tf = read(obj, UInt8)
     tf > 0
 end
-function read(obj::HDF5Object, ::Type{Array{Bool}})
+function read(obj::HDF5Dataset, ::Type{Array{Bool,N} where N})
```
Comment (yuyichao) [inline src/MAT_HDF5.jl]: `Array{Bool}`

Comment (jaakkor2) [reply src/MAT_HDF5.jl]: updated

**Comment by jaakkor2** [comment]
Latest change requests from @yuyichao taken into account.
Tests pass on Linux, random fails in osx.  Local tests pass on Windows x64 Julia v1.1.
Opened https://github.com/JuliaIO/MAT.jl/issues/112 to track the randomly failing osx travis jobs.
For now, should the osx Travis testing be disabled?

**Comment by jaakkor2** [comment]
Close and reopen in hope to trigger another Travis build job

**Comment by timholy** [comment]
Nice to see this getting over the finish line. I cleaned this up, git history-wise, in #113. We can close this when that merges.

**Comment by timholy** [comment]
Thanks for your efforts, @jaakkor2 and @yuyichao!

**Pull request closed without merging**